R/module-visualize-network-server.R

```diff
@@ -0,0 +1,176 @@
+networkServer <- function(input, output, session, parent_session, significant) {
+
+    # Reactive for uploaded CSV data
+    df <- reactive({
+        if (is.null(input$dataUpload) && !is.null(significant())) {
+            df <- significant()
+            if (!is.null(df) && "Protein" %in% names(df)) {
+                df$Protein <- as.character(significant()$Protein)
+                return(df)
+            }
+        }
+        req(input$dataUpload)
+        tryCatch({
+            read.csv(input$dataUpload$datapath)
+        }, error = function(e) {
+            showNotification(paste("Error reading file:", e$message), type = "error")
+            return(NULL)
+        })
+    })
+
+    # Reactive values for input fields with default values
+    proteinIdType <- reactive({
+        req(input$proteinIdType)
+        input$proteinIdType
+    })
+
+    pValue <- reactive({
+        req(input$pValue)
+        as.numeric(input$pValue)
+    })
+
+    logFC <- reactive({
+        req(input$logFC)
+        as.numeric(input$logFC)
+    })
+
+    # Create a reactive expression to generate the JavaScript code for Cytoscape
+    renderNetwork <- reactive({
+        # Annotate protein info and filter the subnetwork based on p-value
+        annotated_df <- tryCatch({
+                annotateProteinInfoFromIndra(df(), proteinIdType())
```

Comment:
Did you add a dependency on MSstatsBioNet for this package? i.e. using the @importFrom tag and adding the dependency in the DESCRIPTION file.